### DISCUSSION TITLE
Built-in CSS autoprefixing based on Browserslist config

### DISCUSSION THREAD

**Discussion by markdalgleish** [remix-run/remix#5017] [Proposals]
Right now, if you want to use [Autoprefixer,](https://github.com/postcss/autoprefixer) you need to [run a separate PostCSS step ahead of time](https://remix.run/docs/en/v1/guides/styling#postcss) which is a bit clunky. It would be nice if this was built-in using your project's [Browserslist](https://github.com/browserslist/browserslist) config via `.browserslistrc` or the `"browserslist"` field in `package.json`.

Another nice benefit of being built-in is that we could choose to use alternative tools like [Lightning CSS.](https://lightningcss.dev)

This would need to be opt-in so it's not a breaking change, but I can imagine this becoming the default behaviour in a future major release.

## Related issues/RFCs/proposals

- https://github.com/remix-run/remix/discussions/4769
- https://github.com/remix-run/remix/discussions/5015

**Comment by sergiodxa** [comment]
In favor of this, also would be great to minify the code.

I used to author my custom CSS in `app/styles` and recently had to move them to `styles` to run PostCSS just to minify the code, would be great to let Remix handle that for any stylesheet I import on my app code.

I also like the idea to support configurations for this, something like
```js
// remix.config.js
module.exports = {
  css: {
    minify: true,
    autoprefix: true,
  },
};
```

**Comment by davyrolink** [comment]
The possibility to let [Lightning CSS](https://lightningcss.dev/) run on the generated css-bundle would be great! That would make it possible, for example, to use [CSS Nesting](https://www.w3.org/TR/css-nesting-1/) syntax in CSS Modules.

**Reply by smorimoto** [reply]
I strongly agree with this.

**Comment by machour** [comment]
This got discussed in [Roadmap #5, around 38'46](https://www.youtube.com/live/tdVFZidXGZo?feature=share&t=2326) and accepted.

Tracked in #5315

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
